pkg/apis/core/validation/validation.go

```diff
@@ -6762,8 +6863,16 @@ func validateBasicResource(quantity resource.Quantity, fldPath *field.Path) fiel
 	return field.ErrorList{}
 }
 
+func ValidatePodResourceRequirements(requirements *core.ResourceRequirements, podClaimNames sets.Set[string], fldPath *field.Path, opts PodValidationOptions) field.ErrorList {
+	return ValidateResourceRequirements(requirements, validatePodResourceName, podClaimNames, fldPath, opts)
+}
+
+func ValidateContainerResourceRequirements(requirements *core.ResourceRequirements, podClaimNames sets.Set[string], fldPath *field.Path, opts PodValidationOptions) field.ErrorList {
+	return ValidateResourceRequirements(requirements, validateContainerResourceName, podClaimNames, fldPath, opts)
+}
+
 // Validates resource requirement spec.
-func ValidateResourceRequirements(requirements *core.ResourceRequirements, podClaimNames sets.Set[string], fldPath *field.Path, opts PodValidationOptions) field.ErrorList {
+func ValidateResourceRequirements(requirements *core.ResourceRequirements, resourceNameFn func(core.ResourceName, *field.Path) field.ErrorList, podClaimNames sets.Set[string], fldPath *field.Path, opts PodValidationOptions) field.ErrorList {
```

Comment:
super nit, but this doesn't seem to need to be public (or else it would be a breaking change)

Reply:
I thought so too. It was a public method before my change so kept it as-is.

Reply:
The new functions are better as a "public" API (even just within k/k) so I think it can be demoted.  More local is more better! :)